src/librustc_passes/rvalue_promotion.rs

```diff
@@ -200,68 +224,72 @@ impl<'a, 'tcx> CheckCrateVisitor<'a, 'tcx> {
         euv::ExprUseVisitor::new(self, tcx, param_env, &region_scope_tree, self.tables, None)
             .consume_body(body);
 
-        self.visit_expr(&body.value);
+        let _ = self.visit_expr(&body.value);
         self.in_fn = outer_in_fn;
         self.tables = outer_tables;
         self.param_env = outer_param_env;
         self.identity_substs = outer_identity_substs;
     }
 
-    fn visit_stmt(&mut self, stmt: &'tcx hir::Stmt) {
+    fn visit_stmt(&mut self, stmt: &'tcx hir::Stmt) -> Result<Promotable, NotPromotable> {
```

Comment:
If you're going to change the signatures of these functions, I'd also change the names to `check_*` instead of `visit_*`.